src/test/lib/optimizer/strategy/in_expression_rewrite_rule_test.cpp

```diff
@@ -238,99 +245,99 @@ TEST_F(InExpressionRewriteRuleTest, JoinStrategy) {
-    EXPECT_LQP_EQ(result_lqp, expected_lqp);
+    EXPECT_LQP_EQ(_lqp, expected_lqp);
 
-    EXPECT_FLOAT_EQ(cardinality_estimator.estimate_cardinality(result_lqp), 1000.f / 200 * 1);
+    EXPECT_FLOAT_EQ(cardinality_estimator.estimate_cardinality(_lqp), 1000.f / 200 * 1);
   }
 
   {
-    // ExpressionEvaluator for five elements
-    const auto input_lqp = PredicateNode::make(five_element_in_expression, node);
-    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
-    EXPECT_EQ(result_lqp, input_lqp);
+    // ExpressionEvaluator for five elements.
+    _lqp = PredicateNode::make(five_element_in_expression, node);
+    const auto expected_lqp = _lqp->deep_copy();
+    _apply_rule(rule, _lqp);
+    EXPECT_LQP_EQ(_lqp, expected_lqp);
 
     // No cardinality check here, as an IN expression with 5 elements will not be touched (see
     // MAX_ELEMENTS_FOR_DISJUNCTION and MIN_ELEMENTS_FOR_JOIN). These InExpressions are currently not supported by the
     // CardinalityEstimator.
   }
 
   {
-    // ExpressionEvaluator for differing types
-    const auto input_lqp = PredicateNode::make(different_types_on_right_side_expression, node);
-    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
-    EXPECT_EQ(result_lqp, input_lqp);
+    // ExpressionEvaluator for differing types.
+    _lqp = PredicateNode::make(different_types_on_right_side_expression, node);
+    const auto expected_lqp = _lqp->deep_copy();
+    _apply_rule(rule, _lqp);
+    EXPECT_LQP_EQ(_lqp, expected_lqp);
   }
 
   {
-    // Join for 100 elements
-    const auto input_lqp = PredicateNode::make(hundred_element_in_expression, node);
-    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
+    // Join for 100 elements.
+    _lqp = PredicateNode::make(hundred_element_in_expression, node);
+    _apply_rule(rule, _lqp);
 
     const auto column_definitions = TableColumnDefinitions{{"right_values", DataType::Int, false}};
-    auto table = std::make_shared<Table>(column_definitions, TableType::Data);
+    const auto table = std::make_shared<Table>(column_definitions, TableType::Data);
     for (auto index = int32_t{0}; index < 100; ++index) {
       table->append({index});
     }
     const auto static_table_node = StaticTableNode::make(table);
     const auto right_col = lqp_column_(static_table_node, ColumnID{0});
     const auto expected_lqp = JoinNode::make(JoinMode::Semi, equals_(col_a, right_col), node, static_table_node);
 
-    EXPECT_LQP_EQ(result_lqp, expected_lqp);
-    EXPECT_TABLE_EQ_UNORDERED(static_cast<StaticTableNode&>(*result_lqp->right_input()).table, table);
+    EXPECT_LQP_EQ(_lqp, expected_lqp);
+    EXPECT_TABLE_EQ_UNORDERED(static_cast<StaticTableNode&>(*_lqp->right_input()).table, table);
 
-    EXPECT_NEAR(cardinality_estimator.estimate_cardinality(result_lqp), 1000.f / 200 * 100, 10);
+    EXPECT_NEAR(cardinality_estimator.estimate_cardinality(_lqp), 1000.f / 200 * 100, 10);
```

Comment:
Just curious. In the FK PR, we used an epsilon and did the comparison manually. Right?

Reply:
`EXPECT_NEAR` expects an absolute value. In the FK PR, the cardinality depends on the SF and so should the epsilon. But I guess `EXPECT_NEAR(actual, expected, expected * epsilon);` should be fine there.